Gems/AWSGameLift/Code/AWSGameLiftClient/Source/AWSGameLiftClientLocalTicketTracker.cpp

```diff
@@ -0,0 +1,169 @@
+                if (describeMatchmakingOutcome.IsSuccess())
+                {
+                    if (describeMatchmakingOutcome.GetResult().GetTicketList().size() == 1)
+                    {
+                        auto ticket = describeMatchmakingOutcome.GetResult().GetTicketList().front();
+                        if (ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::COMPLETED)
+                        {
+                            m_status = TicketTrackerStatus::Idle;
+                            AZ_TracePrintf(AWSGameLiftClientLocalTicketTrackerName,
+                                "Matchmaking ticket %s is complete.", ticket.GetTicketId().c_str());
+                            RequestPlayerJoinMatch(ticket, playerId);
+                            return;
+                        }
+                        else if (ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::TIMED_OUT ||
+                            ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::FAILED ||
+                            ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::CANCELLED)
+                        {
+                            m_status = TicketTrackerStatus::Idle;
+                            AZ_Error(AWSGameLiftClientLocalTicketTrackerName, false, "Matchmaking ticket %s is not complete, %s",
+                                ticket.GetTicketId().c_str(), ticket.GetStatusReason().c_str());
+                            return;
+                        }
+                        else if (ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::REQUIRES_ACCEPTANCE)
+                        {
+                            // broadcast acceptance requires to player
+                        }
+                        else
+                        {
+                            AZ_TracePrintf(AWSGameLiftClientLocalTicketTrackerName, "Matchmaking ticket %s is processing, %s.",
+                                ticket.GetTicketId().c_str(), ticket.GetStatusReason().c_str());
+                        }
+                    }
+                    else
+                    {
+                        AZ_Error(AWSGameLiftClientLocalTicketTrackerName, false, "Unable to find expected ticket with id %s", ticketId.c_str());
+                    }
+                }
+                else
+                {
+                    AZ_Error(AWSGameLiftClientLocalTicketTrackerName, false, AWSGameLiftErrorMessageTemplate,
+                        describeMatchmakingOutcome.GetError().GetExceptionName().c_str(),
+                        describeMatchmakingOutcome.GetError().GetMessage().c_str());
+                }
+            }
+            else
+            {
+                AZ_Error(AWSGameLiftClientLocalTicketTrackerName, false, AWSGameLiftClientMissingErrorMessage);
```

Comment:
IMHO this seems terminal though - is there a point polling if no client?

Reply:
Based on our current interface, it is private to Gem only, so it is not a good idea to stop tracker independently as there is a matchmaking ticket ongoing, because tracker will get started only when ticket is submitted. So tracker has to be stopped together with StopMatchmaking API.

Later I plan to expose tracker API as public ones, then we are free to stop in this case.